tests/test_synthesize_sql_resources.py

```diff
@@ -0,0 +1,3427 @@
+# encoding: utf-8
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# Author: Kyle Lahnakoski (kyle@lahnakoski.com)
+#
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import unicode_literals
+
+from collections import namedtuple
+from unittest import skip
+
+import pyparsing
+from mo_testing.fuzzytestcase import FuzzyTestCase
+
+from moz_sql_parser import parse
+from moz_sql_parser import sql_parser
+
+from moz_sql_parser import sql_synthesizer
+
+from pprint import pformat
+
+class TestSimple(FuzzyTestCase):
+
+    def get_error_message(self, expected_sql, new_sql, expected_json, new_json):
+        res = """
+SQL:         %s
+Broken SQL:  %s
+
+JSON:
+%s
+Broken JSON:
+%s
+""" % (expected_sql, new_sql, pformat(expected_json), pformat(new_json))
+
+        real = ":".join("{:02x}".format(ord(c)) for c in expected_sql)
+        wrong = ":".join("{:02x}".format(ord(c)) for c in new_sql)
+        res = res + """
+Original SQL in HEX: %s
+Synthetic SQL in HEX: %s
+""" % (real, wrong)
+        return res
+
+    def test_001(self):
+        expected_sql = "SELECT * FROM test1"
+        expected_json = {'from': 'test1', 'select': '*'}
+        new_sql = ""
+        new_json = ""
+        try:
+            new_sql = sql_synthesizer.generate_sql(expected_json)
+            new_json = parse(new_sql)
+            self.assertEqual(new_json,
+                             expected_json,
+                             self.get_error_message(expected_sql,
+                                                    new_sql,
+                                                    expected_json,
+                                                    new_json))
+        except Exception as e:
+            print(self.get_error_message(expected_sql, new_sql, expected_json, new_json))
+            print(e)
+            self.assertTrue(False, self.get_error_message(expected_sql,
+                                                          new_sql,
+                                                          expected_json,
+                                                          new_json))
+
+    def test_002(self):
+        expected_sql = "SELECT * FROM test1, test2"
+        expected_json = {'from': ['test1', 'test2'], 'select': '*'}
+        new_sql = ""
+        new_json = ""
+        try:
+            new_sql = sql_synthesizer.generate_sql(expected_json)
+            new_json = parse(new_sql)
+            self.assertEqual(new_json,
+                             expected_json,
+                             self.get_error_message(expected_sql,
+                                                    new_sql,
+                                                    expected_json,
+                                                    new_json))
+        except Exception as e:
```

Comment:
You do not need catch exceptions:  The test will throw an error and show a stack trace, which is better than `print(e)`.

I like how you parse the generated SQL!  😄